subprojects/diagnostics/src/main/java/org/gradle/api/tasks/diagnostics/DependencyInsightReportTask.java

```diff
@@ -270,53 +322,49 @@ private void assertValidTaskConfiguration(@Nullable Configuration configuration)
         return selectedDependencies;
     }
 
-
-    private static AttributeMatchDetails match(Attribute<?> actualAttribute, Object actualValue, AttributeContainer requestedAttributes) {
+    @SuppressWarnings("unchecked")
+    private AttributeMatchDetails match(Attribute<?> actualAttribute, @Nullable Object actualValue, AttributeContainer requestedAttributes) {
+        AttributesSchemaInternal schema = (AttributesSchemaInternal) getProject().getDependencies().getAttributesSchema();
+        // As far as I could tell, the only schema ever mixed in using withProducer is PreferJavaRuntimeVariant
+        // However, that only adds disambiguation rules, which don't apply here. So this should be sufficient:
```

Comment:
For external dependencies, there is no "producer" schema.

For project dependencies, it's a little different. You could have two schemas, one for the consumer and one for the producer project. The schema used by the attribute matcher will be a combination of those two and will use the consumer's compatibility/disambiguation rules first. For a first approximation, this is probably close to being right.